lib/exd/plugin/hello.ex

```diff
@@ -106,32 +108,36 @@ if Code.ensure_loaded?(:hello) do
     @doc """
     Implements route, see module documentation.
     """
-    def route(context(session_id: id), request(method: method, args: args) = _req, uri) do
-      paths = String.split(args["resource"], "/")
-      resource_len = length(paths)
-      cond do
-        resource_len == 1 ->
-          case :hello_binding.lookup(uri, args["resource"]) do
-            {:error, :not_found} -> {:error, :method_not_found}
-            {:ok, _, name} -> {:ok, name, id}
-          end
-        resource_len == 2 ->
-          res = Enum.filter(:hello_binding.all, fn({_,_,resource, _}) -> resource == args["resource"] end)
-          case res do
-            [] -> {:error, :method_not_found}
-            _ ->  {:ok, args["resource"], id}
-          end
-        resource_len == 3 ->
-          app = Enum.at(paths, 0) |> String.to_atom
-          module_api = Exd.Router.apis(method, args["resource"])
-          case module_api do
-            nil ->
+    def route(context(session_id: id) = ctx, request(method: method, args: args) = req, uri) do
```

Comment:
3 level deepness is not good to have, it is a good point to refactor a function in smaller ones.